src/Map.tsx

```diff
@@ -34,6 +34,14 @@ const Map: React.FC<MapProps> = ({
     zoom: 16
   });
 
+  const [hoveredFeature, setHoveredFeature] = useState(undefined);
+
+  const handleHover = useCallback(({ features }: MapEvent) => {
+    setHoveredFeature(
+      Array.isArray(features) && features.length > 0 ? features[0] : ''
```

Comment:
Instead of setting it to an empty string if it doesn't exist, `undefined` would probably be better.